src/app/sessions/components/transaction-dialog/transaction-dialog.component.ts

```diff
@@ -0,0 +1,78 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material';
+import { SessionCreationSummary } from '../../models/session-creation-summary';
+import { Observable } from 'rxjs/Observable';
+import { switchMap, map, tap } from 'rxjs/operators';
+import { combineLatest } from 'rxjs/observable/combineLatest';
+import { Action, select, Store } from '@ngrx/store';
+import { State } from '../../../app.state';
+import { CommitTransaction, RollbackTransaction } from '../../actions/transaction.action';
+import * as fromSessionIndex from '../../reducers';
+import { Problem } from '../../../problems/models/problem.model';
+import { EntityTransaction } from '../../models/transaction-status.model';
+import { Dictionary } from '@ngrx/entity/src/models';
+import * as fromProblems from '../../../problems/reducers';
+
+@Component({
+  selector: 'app-transaction-dialog',
+  templateUrl: './transaction-dialog.component.html',
+  styleUrls: ['./transaction-dialog.component.scss']
+})
+export class TransactionDialogComponent implements OnInit {
+
+  problems$: Observable<Problem[]>;
+  transactionStatus$: Observable<EntityTransaction>;
+  transacted$: Observable<boolean>;
+  problemsLoaded$: Observable<boolean>;
+  finished$: Observable<boolean>;
+  conflicted$: Observable<boolean>;
+  buttonText$: Observable<string>;
+  okAction: Action;
+
+  transactionId: string;
+
+  constructor(
+      public dialogRef: MatDialogRef<TransactionDialogComponent>,
+      @Inject(MAT_DIALOG_DATA) public data: SessionCreationSummary,
+      private store: Store<State>) {
+      this.problems$ = combineLatest(this.store.pipe(select(fromProblems.getProblems)),
```

Comment:
It's getting crowdy here. We should think about when we should split containers into smaller ones